libraries/ms-common/src/stm32f0xx/spi.c

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "spi.h"
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "gpio.h"
+#include "status.h"
+#include "stm32f0xx.h"
+
+static SPI_TypeDef *spi_periph_map[] = { SPI1, SPI2 };
+
+static GPIOAddress s_cs_pin_map[] = { 0, 0 };
```

Comment:
You might as well define a struct to store the SPI typedef and CS GPIO so you don't need to manually keep this array in sync. This is also where having an enum for number of ports would help.